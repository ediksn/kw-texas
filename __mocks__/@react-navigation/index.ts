jest.mock('@react-navigation/native', () => ({
  createNavigatorFactory: jest.fn(),
  useNavigation: () => ({
    navigate: jest.fn(),
    dispatch: jest.fn(),
    goBack: jest.fn(),
    setOptions: jest.fn()
  }),
  useRoute: jest.fn(),
  useFocusEffect: jest.fn()
}))

jest.mock('@react-navigation/material-top-tabs', () => ({
  createMaterialTopTabNavigator: () => ({
    Navigator: jest.mock,
    Screen: jest.mock
  })
}))

jest.mock('@react-navigation/stack', () => ({
  createStackNavigator: () => ({
    Navigator: jest.mock,
    Screen: jest.mock
  })
}))

jest.mock('@react-navigation/bottom-tabs', () => ({
  createBottomTabNavigator: () => ({
    Navigator: jest.mock,
    Screen: jest.mock
  })
}))

export {}
