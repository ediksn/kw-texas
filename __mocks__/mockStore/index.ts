import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { homeModel, loginModel, promptVideoModel } from '~/store/model'
import { VIDEO_INITIAL_STATE } from '~/store/model/videoModel'
import { VIDEO_INITIAL_STATE as BM_VIDEO_INITIAL_STATE } from '~/store/model/bmModel'
import { UPLOAD_INITIAL_STATE } from '~/store/model/uploadVideoModel'
import { USRPROFILE_INITIAL_STATE } from '~/store/model/usrProfileModel'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
const store = mockStore({
  login: loginModel,
  videos: VIDEO_INITIAL_STATE,
  promptVideos: promptVideoModel,
  library: VIDEO_INITIAL_STATE,
  bookmarked: BM_VIDEO_INITIAL_STATE,
  home: homeModel,
  uploadVideo: UPLOAD_INITIAL_STATE,
  usrProfile: USRPROFILE_INITIAL_STATE
})

export default store
