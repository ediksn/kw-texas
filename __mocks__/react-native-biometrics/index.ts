jest.mock('react-native-biometrics', () => ({
  isSensorAvailable: jest.fn(() => Promise.resolve({ then: jest.fn() }))
}))

export {}
