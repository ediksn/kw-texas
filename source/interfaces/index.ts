export { default as loginInterface } from './loginInterface'
export { default as promptVideoInterface } from './promptVideoInterface'
