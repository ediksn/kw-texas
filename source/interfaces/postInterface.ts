import { UploadImageInterface } from './uploadImageInterface'

export interface Link {
  url: string
}

export type PostType = 'LINK' | 'STANDARD' | 'FILE'

export const POST_TYPES: Record<string, PostType> = {
  LINK: 'LINK',
  FILE: 'FILE',
  STANDARD: 'STANDARD'
}
export interface FormPostInterface {
  group?: string
  text: string
  hasImages?: boolean
  images?: any[]
  links?: string[]
  hasLinks?: boolean
}

export interface PostInterface {
  id: string
  content: string
  type: string
  likedBy: string[]
  likesCount: number
  repliesCount: number
  bookmarksCount: number
  creatorfirstName: string
  creatorLastName: string
  isUserCreatorOfThePost: boolean
  creatorId: string
  creatorPhoto: string
  createdAt: string
  updatedAt: string
  deleted: boolean
  detail: {
    attachments?: UploadImageInterface[]
    links?: Link[]
  }
  pinned: boolean
  userHasAlreadyLiked: boolean
  userHasAlreadyBookmarked: boolean
  userHasAlreadyFlagged: boolean
  groupInfo: {
    id: string
    name: string
  }
}

export interface PostProduceProps {
  type: string
  payload?: {
    data: PostInterface[]
    limit: number
  }
}

export interface PostReducerProps {
  draftState: any
  payload?: any
}
