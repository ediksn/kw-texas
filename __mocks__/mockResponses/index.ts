import moment from 'moment'
import { PostInterface } from '~/interfaces/postInterface'

export const videoResponse = {
  id: 2230,
  title: 'zzz',
  description: 'zzz',
  videoUrl: 'file:///data/user/0/com.connectmobile/cache/Camera/b52e283c-40a5-40d0-9cf9-9a51843ec654.mp4',
  imageUrl: null,
  status: 'FINISHED',
  privacy: 'PUBLIC',
  publishedBy: null,
  liked: false,
  createdAt: '1630700559000',
  likesDetail: { likes: 0, __typename: 'Like' },
  bookmarked: false,
  timeRecorded: 1,
  views: 2,
  agents: [
    {
      id: 556397,
      kwUid: 556397,
      firstName: 'Irene',
      lastName: 'Murphy',
      __typename: 'Agent'
    }
  ],
  creatorHashtags: [],
  scripts: [
    {
      videoUrl: 'https://storage.googleapis.com/script-hub-practice-prod-bc0f/5_Waive_the_Appraisal_OBJ_1_low.mp4',
      videoResponseUrl:
        'https://storage.googleapis.com/script-hub-practice-prod-bc0f/5_Waive_the_Appraisal_RESPONSE_1_low.mp4',
      shortDescription:
        'I am hearing that people are waiving their appraisals and I do not want to do that, what does that even mean?',
      hashtags: [
        {
          id: 9,
          description: 'Appointment',
          tag: 'Appointment',
          __typename: 'Hashtag'
        }
      ],
      topics: [
        {
          id: 1,
          title: 'Lead Generation',
          __typename: 'Topic'
        }
      ],
      __typename: 'Script'
    }
  ],
  formats: [],
  __typename: 'ScriptMeeting'
}

export const profileResponse = {
  id: 532768,
  assigneeId: '556397',
  name: 'Kelle Test Team',
  sortName: 'Kelle Test Team',
  role: 'Buyer Specialist',
  photo: 'https://storage.googleapis.com/attachment-qa-eeac/556398/c5dd00kacb05iv34efp0.jpg',
  isPersonal: false,
  isDefault: true,
  members: [28947, 183149],
  userProfile: {
    photo: 'https://avatar.kwconnect.com/556397.jpeg',
    kw_uid: 556397,
    username: 'kelle2',
    first_name: 'Irene',
    last_name: 'Murphy',
    kw_email: null,
    default_org_id: 532768,
    recovery_email: 'kelle2@kw.com',
    email: 'kelle2@kw.com',
    marketing_email: 'tssv@kw.com',
    kw_start_dt: moment('2019-01-14'),
    re_start_dt: null,
    phone: '512-327-3070',
    mobile_phone: null,
    license: null,
    address_1: '1221 South Mopac Expy',
    address_2: 'Ste 400',
    city: 'Austin',
    country: 'US',
    state: 'TX',
    postal_code: '78746'
  }
}

export const postResponse: PostInterface = {
  id: '6172c50ac8a0fb001c4ddf0a',
  content: 'Great2',
  type: 'FILE',
  likedBy: [],
  likesCount: 0,
  repliesCount: 0,
  bookmarksCount: 0,
  isUserCreatorOfThePost: true,
  creatorId: '60ca46ca877e063418825130',
  creatorfirstName: 'Irene',
  creatorLastName: 'Murphy',
  creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
  createdAt: '2021-10-22T14:04:58.171Z',
  deleted: false,
  pinned: false,
  userHasAlreadyLiked: false,
  userHasAlreadyBookmarked: false,
  groupInfo: {
    id: '60f603104d8d7a001ac9ec79',
    name: 'Aanother Best Test Group 2021-08-30T07:11.000Z'
  },
  detail: {
    links: [
      {
        url: 'https://www.youtube.com/channel/UCDlujZcssCwV4P9XItrhC0A'
      }
    ]
  }
}

export const postsResponse = {
  data: {
    data: {
      getPosts: [
        {
          id: '6193d00ade67e4001bbba16e',
          content:
            '{"blocks":[{"key":"1l4c3","text":"J","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-16T15:36:42.438Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '60f603104d8d7a001ac9ec79', name: 'Aanother Best Test Group 2021-08-30T07:11.000Z' }
        },
        {
          id: '61923b6cee9a52001cc268ec',
          content:
            '{"blocks":[{"key":"8kmh4","text":"Now posr ","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60f5b6cbc08afb001d3ba9bc',
          isUserCreatorOfThePost: true,
          creatorfirstName: 'Michael',
          creatorLastName: 'Reid',
          creatorPhoto: 'https://avatar.kwconnect.com/556398.jpeg',
          createdAt: '2021-11-15T10:50:20.180Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '610a9219d330e7001a0c8627', name: 'test_closed' }
        },
        {
          id: '618c0d7912357f001a886ba3',
          content:
            '{"blocks":[{"key":"dovh7","text":"Check out these new homes!","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'FILE',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-10T18:20:41.473Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '61672a0c51c4f9001b9bec9b', name: 'Demo Market Center' }
        },
        {
          id: '618c0ce1d55ede00191adaab',
          content:
            '{"blocks":[{"key":"ehu3r","text":"Hey, Demo Market Center!","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-10T18:18:09.317Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '61672a0c51c4f9001b9bec9b', name: 'Demo Market Center' }
        },
        {
          id: '618c0b51d55ede00191adaa9',
          content:
            '{"blocks":[{"key":"9892l","text":"Hey, this is a post!","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: ['60ca46ca877e063418825130'],
          likesCount: 1,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-10T18:11:29.901Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '61672a0c51c4f9001b9bec9a', name: 'Keller Williams Realty International' }
        },
        {
          id: '618af740d55ede00191ada36',
          content:
            '{"blocks":[{"key":"q97e","text":"I\'m sharing something!","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-09T22:33:36.127Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '61672a0c51c4f9001b9bec9a', name: 'Keller Williams Realty International' }
        },
        {
          id: '618978615de222001c15c163',
          content:
            '{"blocks":[{"key":"2r209","text":"https://www.youtube.com/watch?v=aV2Dfcz20xQ","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[{"offset":0,"length":43,"key":0}],"data":{}}],"entityMap":{"0":{"type":"LINK","mutability":"MUTABLE","data":{"href":"https://www.youtube.com/watch?v=aV2Dfcz20xQ","rel":"noopener noreferrer","target":"_blank","url":"https://www.youtube.com/watch?v=aV2Dfcz20xQ"}}}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60f5b6cbc08afb001d3ba9bc',
          isUserCreatorOfThePost: true,
          creatorfirstName: 'Michael',
          creatorLastName: 'Reid',
          creatorPhoto: 'https://avatar.kwconnect.com/556398.jpeg',
          createdAt: '2021-11-08T19:20:01.313Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '60f603104d8d7a001ac9ec79', name: 'Aanother Best Test Group 2021-08-30T07:11.000Z' }
        },
        {
          id: '618973876c44c0001e5ba233',
          content:
            '{"blocks":[{"key":"bjpoc","text":"https://www.youtube.com/watch?v=daE2-3T5iiU","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60f5b6cbc08afb001d3ba9bc',
          isUserCreatorOfThePost: true,
          creatorfirstName: 'Michael',
          creatorLastName: 'Reid',
          creatorPhoto: 'https://avatar.kwconnect.com/556398.jpeg',
          createdAt: '2021-11-08T18:59:19.164Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '60f603104d8d7a001ac9ec79', name: 'Aanother Best Test Group 2021-08-30T07:11.000Z' }
        },
        {
          id: '61896dc55de222001c15c161',
          content:
            '{"blocks":[{"key":"eeaik","text":"Hello Irene Murphy ! What\'s going on? #kw","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[{"offset":6,"length":12,"key":0}],"data":{}}],"entityMap":{"0":{"type":"mention","mutability":"SEGMENTED","data":{"mention":{"name":"Irene Murphy","avatar":"https://avatar.kwconnect.com/556397.jpeg","link":"https://avatar.kwconnect.com/556397.jpeg"}}}}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60f5b6cbc08afb001d3ba9bc',
          isUserCreatorOfThePost: true,
          creatorfirstName: 'Michael',
          creatorLastName: 'Reid',
          creatorPhoto: 'https://avatar.kwconnect.com/556398.jpeg',
          createdAt: '2021-11-08T18:34:45.856Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '60f603104d8d7a001ac9ec79', name: 'Aanother Best Test Group 2021-08-30T07:11.000Z' }
        },
        {
          id: '61854d2a77f135001cee788b',
          content:
            '{"blocks":[{"key":"99lm1","text":"r","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'FILE',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-05T15:26:34.237Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '610a9fc5d330e7001a0c868b', name: 'test_public_with_members' }
        }
      ]
    }
  }
}

export const createPostResponse = {
  data: {
    createPost: '6172c50ac8a0fb001c4ddf0a'
  }
}

export const addCommentToPostResponse = {
  data: {
    replyPost: '61b899362f3ef4001a768d59'
  }
}

export const eventResponse = {
  bookmarked: true,
  ends: moment('2021-10-11 19:00:00'),
  id: 857429,
  instructors: [
    {
      firstName: 'Gabi',
      lastName: 'Holmberg'
    }
  ],
  location: 'Online',
  name: 'Online Event for KW Texas Agents',
  price: '0',
  published: true,
  slug: null,
  starts: moment('2021-10-10 11:00:00')
}

export const eventsResponse = {
  data: {
    searchOnlineEvents: [
      {
        id: 857784,
        name: 'AT By API_2021-11-01T07:08:31.613',
        location: 'Online',
        starts: '2021-11-01 08:08:31',
        ends: '2021-11-03 11:18:31',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857788,
        name: 'AT By API_2021-11-01T07:08:36.352',
        location: 'Online',
        starts: '2021-11-01 08:08:36',
        ends: '2021-11-03 11:18:36',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: true,
        published: true
      },
      {
        id: 857789,
        name: 'AT By API_2021-11-01T07:08:36.734',
        location: 'Online',
        starts: '2021-11-01 08:08:36',
        ends: '2021-11-03 11:18:36',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857790,
        name: 'AT By API_2021-11-01T07:08:38.294',
        location: 'Online',
        starts: '2021-11-01 08:08:38',
        ends: '2021-11-03 11:18:38',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: true,
        published: true
      },
      {
        id: 857791,
        name: 'AT By API_2021-11-01T07:08:40.191',
        location: 'Online',
        starts: '2021-11-01 08:08:40',
        ends: '2021-11-03 11:18:40',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857792,
        name: 'AT By API_2021-11-01T07:08:41.628',
        location: 'Online',
        starts: '2021-11-01 08:08:41',
        ends: '2021-11-03 11:18:41',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857488,
        name: 'THAT IS Test5555',
        location: 'Online',
        starts: '2021-11-01 11:00:00',
        ends: '2021-11-01 18:00:00',
        slug: null,
        price: '90',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857277,
        name: 'Recurring event for daily',
        location: 'Online',
        starts: '2021-11-01 23:00:00',
        ends: '2021-11-02 00:00:00',
        slug: null,
        price: '0',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857793,
        name: 'AT By API_2021-11-02T07:08:33.065',
        location: 'Online',
        starts: '2021-11-02 08:08:33',
        ends: '2021-11-04 11:18:33',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857796,
        name: 'AT By API_2021-11-02T07:08:38.056',
        location: 'Online',
        starts: '2021-11-02 08:08:38',
        ends: '2021-11-04 11:18:38',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: true,
        published: true
      }
    ]
  }
}
export const deletePostResponse = {
  data: { deletePost: true }
}

export const addBookmarkPostResponse = {
  data: { toggleBookmarkPost: true }
}

export const addLikePostResponse = {
  data: { toggleLikePost: true }
}

export const loginResponse = {
  id_token: 'id_token'
}

export const idTokenJWT = {
  at_hash: 'at_hash',
  sub: 'sub',
  kwoid: 'kwoid',
  auditTrackingId: 'auditTrackingId',
  iss: 'iss',
  tokenName: 'id_token',
  preferred_username: 'preferred_username',
  given_name: 'given_name',
  aud: 'aud',
  azp: 'azp',
  auth_time: 1634927734,
  name: 'name',
  realm: 'realm',
  exp: 1634931334,
  tokenType: 'tokenType',
  family_name: 'family_name',
  iat: 1634927734,
  email: 'email@kw.com',
  kwuid: 'kwuid'
}

export const pickerResponse = [
  {
    type: 'image/jpg',
    fileName: 'image1.jpg',
    width: 3000,
    height: 2002,
    uri: 'http://placehold.it/120x120&text=image1',
    fileSize: 6581432
  },
  {
    type: 'image/jpg',
    fileName: 'image2.jpg',
    width: 3000,
    height: 2002,
    uri: 'http://placehold.it/120x120&text=image2',
    fileSize: 6581432
  }
]
