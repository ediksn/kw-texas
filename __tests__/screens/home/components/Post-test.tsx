import React from 'react'
import renderer from 'react-test-renderer'
import { postResponse } from '../../../../__mocks__/mockResponses'
import Post from '~/components/Post'
import { PostInterface } from '~/interfaces/postInterface'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

test('renders correctly', () => {
  const item: PostInterface = postResponse
  const post = renderer.create(<Post post={item} />).toJSON()
  expect(post).toMatchSnapshot()
})
