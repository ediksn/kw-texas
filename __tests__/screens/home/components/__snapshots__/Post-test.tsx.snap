// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "elevation": 4,
        "padding": 15.714285714285714,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="center"
      source={
        Object {
          "testUri": "../../../source/components/Post/__mocks__/avatarNinaRoyals.png",
        }
      }
      style={
        Object {
          "height": 117.70588235294117,
          "width": 128.57142857142856,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 4,
          "paddingHorizontal": 15.714285714285714,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        NINA ROYALS
      </Text>
      <Text>
        4 hours ago
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="center"
        source={
          Object {
            "testUri": "../../../assets/images/threeDots.png",
          }
        }
        style={
          Object {
            "height": 51.00588235294117,
            "width": 55.714285714285715,
          }
        }
      />
    </View>
  </View>
  <View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={5}
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean tellus leo, fringilla at elit sed, tincidunt viverra dui. Nullam dignissim mi et tincidunt luctus. Nullam ut eleifend diam. Phasellus faucibus aliquam leo, vitae scelerisque nunc dignissim in. Maecenas vulputate quam id metus efficitur tristique. Ut maximus varius enim eleifend feugiat. Ut at ultricies lorem, eget molestie risus. Nam ultricies tempus sem quis fermentum. Sed a eros ac metus scelerisque hendrerit. Ut nec urna aliquam, sagittis eros non, bibendum risus. Ut congue eleifend velit eu dictum. Praesent eu mi vitae leo pulvinar ullamcorper. Sed tortor felis, tristique vitae lacinia sit amet, pellentesque ut felis. Nulla finibus erat libero, eu sodales metus auctor at. Duis iaculis leo vitae odio mollis, feugiat aliquam erat eleifend. Mauris quis dui at lorem tincidunt viverra nec vel mauris. Pellentesque laoreet dapibus purus, eget vehicula nunc scelerisque quis. Donec ac nibh sed erat consequat sagittis ut ut odio. Curabitur tempor vitae neque vel gravida. Morbi varius ligula nec tellus accumsan, quis pharetra ante dictum. Nulla pharetra ac ex id cursus. Vivamus ut lacinia nunc. Maecenas nec orci risus.
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#4797A5",
            "fontWeight": "bold",
          }
        }
      >
        Show
         More
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 19.61764705882353,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6C727E",
          }
        }
      >
        13 likes
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#6C727E",
              },
              Object {
                "marginRight": 21.428571428571427,
              },
            ]
          }
        >
          5 comments
        </Text>
        <Text
          style={
            Object {
              "color": "#6C727E",
            }
          }
        >
          2 shares
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#444444",
        "height": 1.9617647058823529,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 7.857142857142857,
          "borderWidth": 0,
          "elevation": 0,
          "height": 58.85294117647059,
          "justifyContent": "center",
          "margin": 15.714285714285714,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "height": 47.08235294117647,
              "marginRight": 21.428571428571427,
              "width": 51.42857142857143,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 20,
                },
                Object {
                  "color": "#4797A5",
                },
                Object {
                  "fontFamily": "icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
              },
              Object {
                "color": undefined,
              },
              Object {
                "color": "#4797A5",
              },
              Object {
                "fontSize": 34.57142857142857,
              },
            ]
          }
        >
          Like
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 7.857142857142857,
          "borderWidth": 0,
          "elevation": 0,
          "height": 58.85294117647059,
          "justifyContent": "center",
          "margin": 15.714285714285714,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "height": 47.08235294117647,
              "marginRight": 21.428571428571427,
              "width": 51.42857142857143,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 20,
                },
                Object {
                  "color": "#4797A5",
                },
                Object {
                  "fontFamily": "icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
              },
              Object {
                "color": undefined,
              },
              Object {
                "color": "#4797A5",
              },
              Object {
                "fontSize": 34.57142857142857,
              },
            ]
          }
        >
          Comment
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderRadius": 7.857142857142857,
          "borderWidth": 0,
          "elevation": 0,
          "height": 58.85294117647059,
          "justifyContent": "center",
          "margin": 15.714285714285714,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "height": 47.08235294117647,
              "marginRight": 21.428571428571427,
              "width": 51.42857142857143,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 20,
                },
                Object {
                  "color": "#4797A5",
                },
                Object {
                  "fontFamily": "icomoon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
              },
              Object {
                "color": undefined,
              },
              Object {
                "color": "#4797A5",
              },
              Object {
                "fontSize": 34.57142857142857,
              },
            ]
          }
        >
          Share
        </Text>
      </View>
    </View>
  </View>
</View>
`;
